apps/manager_app/get_device_info.c

```diff
@@ -0,0 +1,150 @@
+ * parties, for a fee or other consideration (including without
+ * limitation fees for hosting or consulting/ support services related
+ * to the Software), a product or service whose value derives, entirely
+ * or substantially, from the functionality of the Software. Any license
+ * notice or attribution required by the License must also include
+ * this Commons Clause License Condition notice.
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+
+#include "controller_level_four.h"
+#include "flash_api.h"
+#include "manager_app.h"
+
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/**
+ * @brief Returns the formatted semantic versioning.
+ *
+ * @param firmware_version
+ * @return bool Status indicating the
+ */
+bool get_firmware_version(common_version_t *firmware_version);
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+
+bool get_firmware_version(common_version_t *firmware_version) {
+  uint32_t version = get_fwVer();
+
+  if (NULL == firmware_version || 0 == version || 0xFFFFFFFF == version) {
+    return false;
+  }
+
+  firmware_version->major = (version >> 24) & 0xFF;
+  firmware_version->minor = (version >> 16) & 0xFF;
+  firmware_version->patch = version & 0xFFFF;
+  return true;
+}
+
+/*****************************************************************************
+ * GLOBAL FUNCTIONS
+ *****************************************************************************/
+
+manager_get_device_info_response_t get_device_info() {
```

Comment:
```suggestion
manager_get_device_info_response_t get_device_info(void) {
```

Reply:
By the way as per our last discussion, we decided to check the query type in every step of the flow, so do you think this should also be done here before the operation starts?

Reply:
moved this logic and added the necessary checks